### ISSUE TITLE
isEmpty() doesn't support numeric model values

### ISSUE THREAD

**Issue by avaragado** [machineboy2045/angular-selectize#110]
I have a selectize component with the required attribute set. This means isEmpty is called on validation to check whether the model has a value or not.

If the model value is numeric (eg set programmatically rather than by user selection), the isEmpty test fails as numbers don't have a length property.

**Comment by PhiLhoSoft** [comment]
Yes, at long least, the lazy `!val.length` should be `val.length === 0`...
